src/InterpreterEngine.cpp

```diff
@@ -994,23 +994,58 @@ RamDomain InterpreterEngine::execute(const InterpreterNode* node, InterpreterCon
         ESAC(UnpackRecord)
 
         CASE(Aggregate)
+            bool shouldRunNested = false;
+
             // get the targeted relation
             const InterpreterRelation& rel = *node->getRelation();
 
             // initialize result
             RamDomain res = 0;
+
+            // Use for calculating mean.
+            std::pair<RamFloat, RamFloat> accumulateMean;
+
             switch (cur.getFunction()) {
                 case AggregateOp::min:
-                    res = MAX_RAM_DOMAIN;
+                    res = MAX_RAM_SIGNED;
+                    break;
+                case AggregateOp::umin:
+                    res = ramBitCast(MAX_RAM_UNSIGNED);
+                    break;
+                case AggregateOp::fmin:
+                    res = ramBitCast(MAX_RAM_FLOAT);
                     break;
+
                 case AggregateOp::max:
-                    res = MIN_RAM_DOMAIN;
+                    res = MIN_RAM_SIGNED;
```

Comment:
`ramBitCast` this too for consistency with other numeric types? We may wish to `newtype` `RamDomain` in the future to for better type checks.